diff --git a/netwerk/protocol/http/moz.build b/netwerk/protocol/http/moz.build
--- a/netwerk/protocol/http/moz.build
+++ b/netwerk/protocol/http/moz.build
@@ -257,3 +257,6 @@
 ]
 
 include("/tools/fuzzing/libfuzzer-config.mozbuild")
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
\ No newline at end of file
diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -4,6 +4,7 @@
 
 // HttpLog.h should generally be included first
 #include "nsHttpHandler.h"
+#include "MaskConfig.hpp"
 
 #include <bitset>
 
@@ -993,6 +994,15 @@
 //-----------------------------------------------------------------------------
 
 const nsCString& nsHttpHandler::UserAgent(bool aShouldResistFingerprinting) {
+  if (auto value = MaskConfig::GetString("headers.User-Agent")) {
+    mUserAgent.Assign(nsCString(value.value().c_str()));
+    return mUserAgent;
+  }
+  if (auto value = MaskConfig::GetString("navigator.userAgent")) {
+    mUserAgent.Assign(nsCString(value.value().c_str()));
+    return mUserAgent;
+  }
+
   if (aShouldResistFingerprinting && !mSpoofedUserAgent.IsEmpty()) {
     LOG(("using spoofed userAgent : %s\n", mSpoofedUserAgent.get()));
     return mSpoofedUserAgent;
@@ -2107,6 +2117,10 @@
   }
 
   MOZ_ASSERT(NS_IsMainThread());
+  if (auto value = MaskConfig::GetString("headers.Accept-Language")) {
+    mAcceptLanguages.Assign(nsCString(value.value().c_str()));
+    return NS_OK;
+  }
 
   mAcceptLanguagesIsDirty = false;
 
@@ -2123,6 +2137,11 @@
 
 nsresult nsHttpHandler::SetAcceptEncodings(const char* aAcceptEncodings,
                                            bool isSecure, bool isDictionary) {
+  nsCString encodingOverride;
+  if (auto value = MaskConfig::GetString("headers.Accept-Encoding")) {
+    encodingOverride.Assign(nsCString(value.value().c_str()));
+    aAcceptEncodings = encodingOverride.get();
+  }
   if (isDictionary) {
     mDictionaryAcceptEncodings = aAcceptEncodings;
   } else if (isSecure) {
