diff --git a/browser/components/preferences/preferences.js b/browser/components/preferences/preferences.js
--- a/browser/components/preferences/preferences.js
+++ b/browser/components/preferences/preferences.js
@@ -639,8 +639,6 @@
   categories.addEventListener("change-view", event => {
     gotoPref(event.target.view);
   });
-
-  maybeDisplayPoliciesNotice();
 
   window.addEventListener("hashchange", onHashChange);
   window.addEventListener("beforeunload", onBeforeunload);
