diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -264,6 +264,11 @@
 #ifdef XP_WIN
 @RESPATH@/browser/components/shell/assets/kit.gif
 #endif
+
+; Camoufox specific files
+@RESPATH@/defaults/pref/local-settings.js
+@RESPATH@/distribution/policies.json
+@RESPATH@/camoufox.cfg
 
 ; [Default Preferences]
 ; All the pref files must be part of base to prevent migration bugs
diff --git a/lw/moz.build b/lw/moz.build
new file mode 100644
--- /dev/null
+++ b/lw/moz.build
@@ -0,0 +1,14 @@
+FINAL_TARGET_FILES += [
+  "camoufox.cfg",
+  "chrome.css",
+  "properties.json"
+]
+
+FINAL_TARGET_FILES.distribution += [
+  "policies.json",
+]
+
+# local-settings does not yet end up being pacakged.
+FINAL_TARGET_FILES.defaults.pref += [
+  "local-settings.js",
+]
diff --git a/moz.build b/moz.build
--- a/moz.build
+++ b/moz.build
@@ -355,3 +355,5 @@
 SPHINX_TREES["ai-agent-tools"] = "docs/ai-agent-tools"
 
 include("build/templates.mozbuild")
+
+DIRS += ["lw"]
