diff --git a/dom/ipc/ContentParent.cpp b/dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp
+++ b/dom/ipc/ContentParent.cpp
@@ -2402,22 +2402,24 @@
     // Repo dir
     nsCOMPtr<nsIFile> repoDir;
     rv = nsMacUtilsImpl::GetRepoDir(getter_AddRefs(repoDir));
-    if (NS_FAILED(rv)) {
-      MOZ_CRASH("Failed to get path to repo dir");
-    }
-    nsCString repoDirPath;
-    (void)repoDir->GetNativePath(repoDirPath);
-    info.testingReadPath3 = repoDirPath.get();
+    if (NS_SUCCEEDED(rv)) {
+      nsCString repoDirPath;
+      (void)repoDir->GetNativePath(repoDirPath);
+      info.testingReadPath3 = repoDirPath.get();
+    } else {
+      NS_WARNING("Failed to get repo dir path for sandbox, skipping testingReadPath3");
+    }
 
     // Object dir
     nsCOMPtr<nsIFile> objDir;
     rv = nsMacUtilsImpl::GetObjDir(getter_AddRefs(objDir));
-    if (NS_FAILED(rv)) {
-      MOZ_CRASH("Failed to get path to build object dir");
-    }
-    nsCString objDirPath;
-    (void)objDir->GetNativePath(objDirPath);
-    info.testingReadPath4 = objDirPath.get();
+    if (NS_SUCCEEDED(rv)) {
+      nsCString objDirPath;
+      (void)objDir->GetNativePath(objDirPath);
+      info.testingReadPath4 = objDirPath.get();
+    } else {
+      NS_WARNING("Failed to get obj dir path for sandbox, skipping testingReadPath4");
+    }
   }
 
   // DEBUG_WRITE_DIR
