diff --git a/browser/app/firefox.exe.manifest b/browser/app/firefox.exe.manifest
--- a/browser/app/firefox.exe.manifest
+++ b/browser/app/firefox.exe.manifest
@@ -3,10 +3,10 @@
 <assemblyIdentity
         version="1.0.0.0"
         processorArchitecture="*"
-        name="Firefox"
+        name="Camoufox"
         type="win32"
 />
-<description>Firefox</description>
+<description>Camoufox</description>
 <dependency>
         <dependentAssembly>
                 <assemblyIdentity
diff --git a/browser/app/moz.build b/browser/app/moz.build
--- a/browser/app/moz.build
+++ b/browser/app/moz.build
@@ -74,8 +74,9 @@
     extra_files = [
         (f"{branding}/firefox.icns", "Resources/firefox.icns"),
         (f"{branding}/document.icns", "Resources/document.icns"),
-        (f"{branding}/Assets.car", "Resources/Assets.car"),
-    ]
+    ]
+    if CONFIG["MOZ_BRANDING_DIRECTORY"] != "browser/branding/camoufox":
+        extra_files.append((f"{branding}/Assets.car", "Resources/Assets.car"))
     if CONFIG["MOZ_UPDATER"]:
         extra_files.append((
             f"!/{FINAL_TARGET}/org.mozilla.updater",
