diff --git a/python/mozversioncontrol/mozversioncontrol/repo/source.py b/python/mozversioncontrol/mozversioncontrol/repo/source.py
--- a/python/mozversioncontrol/mozversioncontrol/repo/source.py
+++ b/python/mozversioncontrol/mozversioncontrol/repo/source.py
@@ -111,7 +111,7 @@
         for root, dirs, files in os.walk(self.path):
             base = os.path.relpath(root, self.path)
             for name in files:
-                res.append(os.path.join(base, name))
+                res.append(os.path.join(base, name).replace("\\", "/"))
         return res
 
     def get_tracked_files_finder(self, path):
