diff --git a/toolkit/components/extensions/parent/ext-storage.js b/toolkit/components/extensions/parent/ext-storage.js
--- a/toolkit/components/extensions/parent/ext-storage.js
+++ b/toolkit/components/extensions/parent/ext-storage.js
@@ -411,6 +411,14 @@
         managed: {
           async get(keys) {
             enforceNoTemporaryAddon(extension.id);
+            const assetsBootstrapLocation = Services.prefs.getStringPref("librewolf.uBO.assetsBootstrapLocation", undefined);
+            if (extension.id == "uBlock0@raymondhill.net" && assetsBootstrapLocation) {
+              return {
+                adminSettings: {
+                  assetsBootstrapLocation
+                }
+              }
+            }
             let data = await getManagedStorageManifestData(extension, context);
             return ExtensionStorage._filterProperties(extension.id, data, keys);
           },
