diff --git a/devtools/server/actors/thread.js b/devtools/server/actors/thread.js
--- a/devtools/server/actors/thread.js
+++ b/devtools/server/actors/thread.js
@@ -390,7 +390,8 @@
   attach(options) {
     // Note that the client avoids trying to call attach if already attached.
     // But just in case, avoid any possible duplicate call to attach.
-    if (this.alreadyAttached) {
+    let forceDetach = Services.prefs.getBoolPref("librewolf.debugger.force_detach", false);
+    if (this.alreadyAttached || forceDetach) {
       return;
     }
 
diff --git a/devtools/server/actors/webconsole/listeners/console-api.js b/devtools/server/actors/webconsole/listeners/console-api.js
--- a/devtools/server/actors/webconsole/listeners/console-api.js
+++ b/devtools/server/actors/webconsole/listeners/console-api.js
@@ -98,7 +98,8 @@
    *        The message object receives from the ConsoleAPIStorage service.
    */
   onConsoleAPILogEvent(message) {
-    if (!this.handler) {
+    let disableConsole = Services.prefs.getBoolPref("librewolf.console.logging_disabled", false);
+    if (!this.handler || disableConsole) {
       return;
     }
 
