diff --git a/browser/components/urlbar/UrlbarProviderInterventions.sys.mjs b/browser/components/urlbar/UrlbarProviderInterventions.sys.mjs
index 575205158bb8..e0c76c682a7d 100644
--- a/browser/components/urlbar/UrlbarProviderInterventions.sys.mjs
+++ b/browser/components/urlbar/UrlbarProviderInterventions.sys.mjs
@@ -454,7 +454,7 @@ class ProviderInterventions extends UrlbarProvider {
         ]),
       });
       for (let [id, phrases] of Object.entries(DOCUMENTS)) {
-        queryScorer.addDocument({ id, phrases });
+//        queryScorer.addDocument({ id, phrases });
       }
       return queryScorer;
     });
@@ -494,7 +494,7 @@ class ProviderInterventions extends UrlbarProvider {
    * @returns {boolean} Whether this provider should be invoked for the search.
    */
   isActive(queryContext) {
-    if (
+    if (true ||
       !queryContext.searchString ||
       queryContext.searchString.length > UrlbarUtils.MAX_TEXT_LENGTH ||
       lazy.UrlbarTokenizer.REGEXP_LIKE_PROTOCOL.test(
