diff --git a/dom/media/CubebUtils.cpp b/dom/media/CubebUtils.cpp
--- a/dom/media/CubebUtils.cpp
+++ b/dom/media/CubebUtils.cpp
@@ -46,6 +46,7 @@
 #include "audioipc2_client_ffi_generated.h"
 #include "audioipc2_server_ffi_generated.h"
 #include "mozilla/StaticPrefs_media.h"
+#include "MaskConfig.hpp"
 
 #if defined(ENABLE_TESTS) || defined(FUZZING)
 #  define ENABLE_MOCK_CUBEB 1
@@ -501,6 +502,8 @@
   if (sCubebForcedSampleRate) {
     return sCubebForcedSampleRate;
   }
+  if (auto value = MaskConfig::GetUint32("AudioContext:sampleRate"))
+    return value.value();
   if (aShouldResistFingerprinting) {
     return 44100;
   }
diff --git a/dom/media/moz.build b/dom/media/moz.build
--- a/dom/media/moz.build
+++ b/dom/media/moz.build
@@ -436,5 +436,7 @@
 COMPILE_FLAGS["WARNINGS_CXXFLAGS"] += ["-Werror=switch"]
 
 FINAL_LIBRARY = "xul"
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
 
 SPHINX_TREES["/media"] = "docs"
diff --git a/dom/media/webaudio/AudioContext.cpp b/dom/media/webaudio/AudioContext.cpp
--- a/dom/media/webaudio/AudioContext.cpp
+++ b/dom/media/webaudio/AudioContext.cpp
@@ -4,6 +4,7 @@
 
 #include "AudioContext.h"
 
+#include "MaskConfig.hpp"
 #include "AudioBuffer.h"
 #include "AudioBufferSourceNode.h"
 #include "AudioChannelService.h"
@@ -560,6 +561,8 @@
   }
   // When reduceFingerprinting is enabled, return a latency figure that is
   // fixed, but plausible for the platform.
+  if (auto value = MaskConfig::GetDouble("AudioContext:outputLatency"))
+    return value.value();
   double latency_s = 0.0;
   if (mShouldResistFingerprinting) {
 #ifdef XP_MACOSX
@@ -713,6 +716,8 @@
 }
 
 uint32_t AudioContext::MaxChannelCount() const {
+  if (auto value = MaskConfig::GetUint32("AudioContext:maxChannelCount"))
+    return value.value();
   if (mShouldResistFingerprinting) {
     return 2;
   }
diff --git a/dom/media/webaudio/moz.build b/dom/media/webaudio/moz.build
--- a/dom/media/webaudio/moz.build
+++ b/dom/media/webaudio/moz.build
@@ -149,3 +149,6 @@
 
 FINAL_LIBRARY = "xul"
 LOCAL_INCLUDES += [".."]
+
+# DOM Mask
+LOCAL_INCLUDES += ["/camoucfg"]
