diff --git a/browser/components/extensions/parent/ext-browser.js b/browser/components/extensions/parent/ext-browser.js
--- a/browser/components/extensions/parent/ext-browser.js
+++ b/browser/components/extensions/parent/ext-browser.js
@@ -66,6 +66,9 @@
 /* eslint-enable mozilla/balanced-listeners */
 
 global.openOptionsPage = extension => {
+  if (!ChromeUtils.camouGetBool('allowAddonNewtab', false)) {
+    return Promise.reject({ message: "Rejected by Camoufox." });
+  }
   let window = windowTracker.topWindow;
   if (!window) {
     return Promise.reject({ message: "No browser window available" });
diff --git a/browser/components/extensions/parent/ext-tabs.js b/browser/components/extensions/parent/ext-tabs.js
--- a/browser/components/extensions/parent/ext-tabs.js
+++ b/browser/components/extensions/parent/ext-tabs.js
@@ -689,6 +689,9 @@
         }).api(),
 
         create(createProperties) {
+          if (!ChromeUtils.camouGetBool('allowAddonNewtab', false)) {
+            return Promise.reject({ message: "Rejected by Camoufox." });
+          }
           return new Promise(resolve => {
             let window =
               createProperties.windowId !== null
