
diff --git a/toolkit/components/search/SearchEngineSelector.sys.mjs b/toolkit/components/search/SearchEngineSelector.sys.mjs
index 3dab73ee2f..c5c1259217 100644
--- a/toolkit/components/search/SearchEngineSelector.sys.mjs
+++ b/toolkit/components/search/SearchEngineSelector.sys.mjs
@@ -158,6 +158,21 @@ export class SearchEngineSelector {
    *   could be obtained.
    */
   async _getConfiguration(firstTime = true) {
+    if (true) {
+      return [
+        {
+          "appliesTo": [{
+            "default": "yes",
+            "included": {
+              "everywhere": true
+            },
+            "webExtension": {
+              "id": "none@mozilla.org"
+            }
+          }],
+        },
+      ];
+    }
     let result = [];
     let failed = false;
     try {
