diff --git a/browser/base/content/browser-addons.js b/browser/base/content/browser-addons.js
--- a/browser/base/content/browser-addons.js
+++ b/browser/base/content/browser-addons.js
@@ -3091,6 +3091,8 @@
   // class and not the `.toolbarbutton-1` one. When NOT placed in the panel,
   // it is the other way around.
   _updateWidgetClassName(aWidgetId, inPanel) {
+    inPanel = false;
+    this.pinToToolbar(aWidgetId, true);
     if (!CustomizableUI.isWebExtensionWidget(aWidgetId)) {
       return;
     }
diff --git a/browser/base/content/main-popupset.inc.xhtml b/browser/base/content/main-popupset.inc.xhtml
--- a/browser/base/content/main-popupset.inc.xhtml
+++ b/browser/base/content/main-popupset.inc.xhtml
@@ -621,7 +621,6 @@
               type="checkbox"
               contexttype="toolbaritem"
               class="customize-context-pinToToolbar"/>
-    <menuseparator id="toolbarDownloadsAnchorMenuSeparator"/>
     <menuitem id="toolbar-context-always-open-downloads-panel"
               type="checkbox"
               data-lazy-l10n-id="toolbar-context-menu-always-open-downloads-panel"
