diff --git a/dom/webidl/Element.webidl b/dom/webidl/Element.webidl
--- a/dom/webidl/Element.webidl
+++ b/dom/webidl/Element.webidl
@@ -295,6 +295,9 @@
   [BinaryName="shadowRootForBindings"]
   readonly attribute ShadowRoot? shadowRoot;
 
+  [Func="Document::IsCallerChromeOrAddon", BinaryName="shadowRoot"]
+  readonly attribute ShadowRoot? shadowRootUnl;
+
   [NeedsSubjectPrincipal, Func="Document::IsCallerChromeOrAddon"]
   readonly attribute ShadowRoot? openOrClosedShadowRoot;
 
