diff --git a/dom/webidl/Element.webidl b/dom/webidl/Element.webidl
--- a/dom/webidl/Element.webidl
+++ b/dom/webidl/Element.webidl
@@ -291,6 +291,9 @@ partial interface Element {
   ShadowRoot attachShadow(ShadowRootInit shadowRootInitDict);
   [BinaryName="shadowRootForBindings"]
   readonly attribute ShadowRoot? shadowRoot;
 
+  [Func="Document::IsCallerChromeOrAddon", BinaryName="shadowRoot"]
+  readonly attribute ShadowRoot? shadowRootUnl;
+
   [NeedsSubjectPrincipal, Func="Document::IsCallerChromeOrAddon"]
   readonly attribute ShadowRoot? openOrClosedShadowRoot;
